File: GoC/foo.c
/*
 * gcc -DUSE_MAIN foo.c -o foo.exe
 */

#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include "foo.h"

/*
 * void -> void
 */
void helloworld() {
    printf("Hello, world.\n");
}

/*
 * int -> int
 */
int square(int x) {
    return x*x;
}

/*
 * (int, int) -> int
 */
int add(int x, int y) {
    return x+y;
}

/*
 * string -> string
 */
char* make_greet(char *x) {
    char *r = (char*)malloc(10+strlen(x));
    sprintf(r, "Hello, %s!", x);
    return r;
}

void free_greet(char *x) {
    free(x);
}

/*
 * (int,int) -> (int, int)
 */
MyDiv* mydiv(int x, int y) {
    MyDiv *r = (MyDiv*)malloc(sizeof(MyDiv));
    r->answer = x/y;
    r->remainder = x%y;
    return r;
}

int mydiv_answer(MyDiv* x) {
    return x->answer;
}

int mydiv_remainder(MyDiv* x) {
    return x->remainder;
}

void mydiv_free(MyDiv *x) {
    free(x);
}

#ifdef USE_MAIN
int main(int argc, char**argv) {
    int x, y;
    char *name, *msg;
    MyDiv* r;

    helloworld();

    x = 111;
    printf("square(%d) = %d\n", x, square(x));

    x = 123;
    y = 654;
    printf("add(%d, %d) = %d\n", x, y, add(x, y));

    name = "DARK STAR";
    msg = make_greet(name);
    printf("%s\n", msg);
    free_greet(msg);

    x = 1000;
    y = 17;
    r = mydiv(x, y);
    printf("%d / %d = %d ... %d\n", x, y, mydiv_answer(r),
      mydiv_remainder(r));
    mydiv_free(r);
    return 0;
}
#endif /* USE_MAIN */

File: GoPy/foo.py
# -*- coding: utf-8 -*-

# void -> void
def helloworld():
    print "Hello, world."

# int -> int
def square(x):
    return x*x

# (int, int) -> int
def myadd(x, y):
    return x+y

# string -> string
def make_greet(x):
    return "Hello, %s!" % (x)

# (int,int) -> (int, int)
def mydiv(x, y):
    answer = x/y
    remainder = x%y
    return answer, remainder

def main():
    helloworld()
    x = 111
    print "square(%d) = %d" % (x, square(x))
    x, y = 123, 654
    print "add(%d, %d) = %d" % (x, y, myadd(x, y))
    name = "DARK STAR"
    print make_greet(name)
    x, y = 1000, 17
    ans, rem = mydiv(x, y)
    print "%d / %d = %d ... %d" % (x, y, ans, rem)

#if __name__ == "__main__":
#    main()